modules/api/php/endpoint.class.inc

```diff
@@ -72,4 +72,33 @@ abstract class Endpoint implements RequestHandlerInterface
 
         return $handler->handle($request);
     }
+
+    /**
+     * Used to create (select, text, ...) element for form filter.
+     *
+     * @param string     $name   for element
+     * @param string     $type   for element
+     * @param string     $label  for element
+     * @param bool       $hidden for element
+     * @param array|null $values for element
+     *
+     * @return object
+     */
+    public function formElementObjectCreator(
```

Comment:
This do not belong in the api/endpoint class. I see that this function is used by the subclasses but having a specific class to manage the response "formating" (like a view) would be more appropriate (imo). The subclass could create a LorisForm, add elements and then pass the LorisForm to the view class for it to arrange it.

Reply:
This isn't resolved, this still doesn't belong in the API.